### PR TITLE
[Falcon7b] Add top-1/top-5 metrics to perplexity test and add 128/2K seq lengths

### PR DESCRIPTION
**Pull request by skhorasganiTT**
### Ticket
- #5383

### Problem description
- 128/2k seq lengths and top1/5 metrics were missing from the falcon7b perplexity test

### What's changed
- Added 128/2k seq lengths for prefill/decode and huggingface/metal models
- Added top1/5 metrics and updated target verification
- Created utility functions for metric calculation and verification in llm_dataset_utils.py
- Re-enabled perplexity test for 1k decode in ci due to #9268 being fixed
- (todo in later PR): add ci tests for 128/2k seq length and move 1k seq tests once there is a pipeline for accuracy benchmarking

### Checklist
- [x] Post commit CI passes (not applicable, changes contained to model)
- [x] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by skhorasganiTT** [comment]
T3K demo tests - https://github.com/tenstorrent/tt-metal/actions/runs/9507864194/job/26208268455

**Review by pavlepopovic — APPROVED** [review]

**Review by pavlejosipovic — APPROVED** [review]

**Review by s-jovic — APPROVED** [review]

File: models/datasets/llm_dataset_utils.py [hunk 1]
```diff
@@ -80,3 +83,34 @@ def prepare_textgen_dataloader(
...
+    # Calculate top-1/5 accuracy
+    logits = logits.float().detach().numpy()
+    labels = labels.float().detach().numpy()
+    top1 = top_k_accuracy_score(labels, logits, k=1, labels=np.arange(logits.shape[-1]))
+    top5 = top_k_accuracy_score(labels, logits, k=5, labels=np.arange(logits.shape[-1]))
+
+    return nll, top1, top5
+
+
+def verify_acc_metrics(ppl, top1_acc, top5_acc, expected_ppl_top1_top5):
```
Comment (djordje-tt) [inline models/datasets/llm_dataset_utils.py:103]: I'd recommend passing either dictionary for both expected and calculated arguments or to pass 6 separate arguments. With dictionaries, you can compare values of the same keys (different metrics) and this way you are relying on the fact that you are always passing args in the same order in list expected_ppl_top1_top5. And I think it would better be named expected_acc_metrics or something similar.

**Review by boris-drazic — APPROVED** [review]

File: models/datasets/llm_dataset_utils.py (hunk 1 above)
Comment (skhorasganiTT) [reply models/datasets/llm_dataset_utils.py]: Agreed, fixed

File: tests/scripts/t3000/run_t3000_demo_tests.sh [hunk 2]
```diff
@@ -31,8 +31,8 @@ run_t3000_falcon7b_tests(){
   WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest --disable-warnings -q -s --input-method=json --input-path='models/demos/t3000/falcon7b/input_data_t3000.json' models/demos/t3000/falcon7b/demo_t3000.py::test_demo_multichip[user_input0-8-True-default_mode_greedy_verify]
 
   # Falcon7B perplexity test (prefill and decode)
-  WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest models/demos/falcon7b/tests/test_perplexity_falcon.py::test_perplexity[1-True-prefill_seq1024_dram]
-  # WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest models/demos/falcon7b/tests/test_perplexity_falcon.py::test_perplexity[1-True-decode_1024_l1_sharded]  # Disabled due to Issue #9268
+  WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest models/demos/falcon7b/tests/test_perplexity_falcon.py::test_perplexity[True-prefill_seq1024_dram]
+  WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest models/demos/falcon7b/tests/test_perplexity_falcon.py::test_perplexity[True-decode_1024_l1_sharded]
```
Comment (ttmchiou) [inline tests/scripts/t3000/run_t3000_demo_tests.sh:35]: Noticing in the raw logs that there is a deprecation warning.
Should this be patched in this PR too?

Workflow link: https://github.com/tenstorrent/tt-metal/actions/runs/9519157120/job/26241907392

Raw Log link: https://productionresultssa11.blob.core.windows.net/actions-results/7e9c6ce2-3d4f-4394-b175-1fae85eaa3f4/workflow-job-run-b58f06d6-9baa-5eb3-9035-03009dd223b5/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-06-15T00%3A48%3A31Z&sig=vJxFRrchm9hR51HKSMS68ESeUJA8EylWFYTZvfD22U0%3D&ske=2024-06-15T09%3A34%3A35Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-06-14T21%3A34%3A35Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2023-11-03&sp=r&spr=https&sr=b&st=2024-06-15T00%3A38%3A26Z&sv=2023-11-03
```
2024-06-14T17:01:39.2716048Z models/demos/falcon7b/tests/test_perplexity_falcon.py::test_perplexity[True-prefill_seq1024_dram]
2024-06-14T17:01:39.2717994Z   /home/ubuntu/actions-runner/_work/tt-metal/tt-metal/python_env/lib/python3.8/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
```

**Review by djordje-tt — APPROVED** [review]

**Review by ttmchiou — APPROVED** [review]
